docs/source/cost.rst

```diff
@@ -0,0 +1,329 @@
+and implemented for deterministic forecasts by
+:py:func:`solarforecastarbiter.metrics.deterministic.datetime_cost`. Similar
+to the time of day cost, the datamodel expects `aggregation`, `net`,
+and `fill` parameters. In this case `cost` values are associated with
+each date-time specified in `datetimes`. The `timezone` parameter
+defines the timezone if `datetimes` are not localized, and if
+`timezone` is None, the timezone of the errors is used.
+
+The minimum/maximum bounds of `datetimes` should cover the range of
+date-times that one wants to evaluate. For example, when evaluating
+the cost defined by
+
+
+.. code-block:: python
+
+    import datetime
+    from solarforecastarbiter import datamodel
+
+    cost_model = datamodel.DatetimeCost(
+        cost=[1.3, 1.9, 0.9, 2.0],
+        times=[datetime.datetime(2020, 5, 1, 12, 0),
+               datetime.datetime(2020, 5, 2, 12, 0),
+               datetime.datetime(2020, 5, 3, 12, 0),
+               datetime.datetime(2020, 5, 4, 12, 0)],
+        net=True,
+        aggregation='sum',
+        fill='forward',
+        timezone='UTC'
+    )
+
+errors in the timeseries before 2020-05-01T12:00 are not included in
+the final calculation.
+
+
+Error Band Cost
+===============
+
+The error band cost model is defined using :py:class:`.ErrorBandCost`
+and implemented for deterministic forecasts by
+:py:func:`solarforecastarbiter.metrics.deterministic.error_band_cost`.
+Each of `bands` is a :py:class:`.CostBand` that describes the range of
+errors the band applies to and the parameters for one of the costs
+above. For example,
+
+.. code-block:: python
+
+    import datetime
+    from solarforecastarbiter import datamodel
+
+    cost_model = datamodel.ErrorBandCost(
+        bands=[
+            datamodel.CostBand(
+                error_range=(-5.0, 20.5),
+                cost_function='constant',
+                cost_function_parameters=datamodel.ConstantCost(
+                    cost=33.0,
+                    net=True,
+                    aggregation='sum'
+                )
+            ),
+            datamodel.CostBand(
+                error_range=(20.5, float('inf')),
+                cost_function='timeofday'
+                cost_function_parameters=datamodel.TimeOfDayCost(
+                    cost=[3.3, 1.2],
+                    times=[datetime.time(hour=15), datetime.time(hour=20)],
+                    net=True,
+                    aggregation='sum',
+                    fill='forward'
+                )
+            )
+        ]
+    )
+
+defines a cost that will apply a constant cost of $33.0 / unit error
+for all errors in the range [-5.0, 20.5]. For errors > 20.5, the time
+of day cost applies. The errors within each band are aggregated
+according to the `aggregation` and `net` parameter of the band
+parameters, but the total cost is the sum of all error bands.
+
+Band error ranges are evaluated in order and any errors outside the
```

Comment:
in order they are specified or in order of minimum band value?

Reply:
specified

Reply:
will clarify